### PR TITLE
fix: minor bug fixes

### PR DESCRIPTION
**Pull request by rililive**

### CODE REVIEW

File: components/Products.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ const Products = ({ productData }: any) => {
                 {item.title.substring(0, 25)}
               </p>
               <p className="text-base text-zinc-500">
-                {item.description.substring(0, 80)}...
+                {item.description?.substring(0, 80)}...
```
Comment (rililive) [inline components/Products.tsx:96]: This way if `item.description` is undefined, things don't blow up

File: components/RestaurantCard.tsx [hunk 2]
```diff
@@ -53,6 +55,9 @@ const RestaurantCard = ({ restaurant }: any) => {
     <div
       key={restaurant.name}
       className="border-[1px] w-full h-full border-gray-200 mb-2 group shadow-md rounded-xl"
+      onClick={() => {
+        router.push(`restaurants/${restaurant.name}`)
+      }}
```
Comment (rililive) [inline components/RestaurantCard.tsx:60]: It wasn't obvious that the user needs to click the `+ menu` button, so this navigates to the restaurant page if they click anywhere on the preview tile

File: redux/shoppersSlice.ts [hunk 3]
```diff
@@ -54,6 +54,15 @@ export const shopperslice = createSlice({
...
     },
+    updateItem: (state, action) => {
+      const itemIndex = state.productData.findIndex(
+        (item: StoreProduct) => item.id === action.payload._id
+      );
+      state.productData[itemIndex] = {
+        ...state.productData[itemIndex],
+        ...action.payload,
+      }
+    },
```
Comment (rililive) [inline redux/shoppersSlice.ts:65]: Not sure if this is necessary, but I noticed you were trying to modify an "Item" in a map. With redux you should only ever modify a state object within the reducers here (not within the actual React component)

File: components/RestaurantCard.tsx (hunk 2 above)
Comment (BlueRacoon) [reply components/RestaurantCard.tsx:60]: Good idea - I like the feedback on UI stuff like this

File: redux/shoppersSlice.ts (hunk 3 above)
Comment (BlueRacoon) [reply redux/shoppersSlice.ts:65]: Gotcha - still learning Redux but that is good to know. I'll try to be aware of this and not modify states directly from a component.

**Review by BlueRacoon — APPROVED** [review]
Looks good - we probably will not use the products.tsx, but that's my fault for leaving unused code in the repo lol.

**Pull request merged**